WordPress/gutenberg
# Github workflow: add a PHP backport changes action

## What?

This is just a proposal...

Adds a Github action that:
- detects whether any PHP files have been added or changed in `/lib`, `/phpunit` or ~`block-library`~ directories
- creates a comment on a PR to inquire about if those PHP changes required backporting
- updates that comment as PHP changes are detected on push
- updates the comment if it no longer finds PHP changes


<img width="956" alt="Screenshot 2023-06-30 at 10 58 39 am" src="https://github.com/WordPress/gutenberg/assets/6458278/4843ba41-16e8-4b78-b940-65c5edf99291">

<img width="936" alt="Screenshot 2023-06-30 at 11 24 25 am" src="https://github.com/WordPress/gutenberg/assets/6458278/3bfc61b9-531f-4b76-9e8f-ba671b47c91f">



Nice to haves as a follow up (will probably require a custom Gutenberg action as the marketplace is a desert)
- remove the comment when no PHP files are detected instead of updating it
- add a "Needs PHP backport" label tentatively ? Might be an overkill as we can't be sure the PHP files require it

## Why?
WordPress releases are hectic and would be made easier if plugin changes that are destined for Core are reviewed and tested against Core earlier.

## How?
Adding a new github action YAML file.

## Testing Instructions

- Push some PHP changes in either the `/phpunit`, `/lib` or `packages/block-library/src` folders to this branch (just edit comments if you like) and see if the message is appears.
- Make sure the PHP list of changes is updated in the comment as new additions/changes are pushed
